### PR TITLE
RWD: New design for star rating

### PR DESCRIPTION
**Pull request by empiricompany**
I wanted to share and propose this new and more modern way to display the stars in the review form.
It's all done only with CSS!

demo-review-rating-star

P.S.: I also changed the colors to a more standard yellow (the same of backend) for rating stars.

### CODE REVIEW

**Comment by fballiano** [comment]
Loving this PR @empiricompany!

Do you think the color has enough contrast on a white background? to my eyes it's a bit hard to see and I don't know if I like it that much:
Screenshot 2024-12-17 alle 20 32 47



Contrast analyzer doesn't pass either (but ok this is not text..)
Screenshot 2024-12-17 alle 20 34 26

File: public/skin/frontend/rwd/default/css/styles.css [hunk 1]
```diff
@@ -1604,6 +1604,50 @@ span.weee {
...
+  } 
+  .star-rating__value:checked + .star-rating__label:hover,
+  .star-rating__value:checked + .star-rating__label:hover ~ .star-rating__label,
+  .star-rating__value:checked ~ .star-rating__label:hover,
+  .star-rating__value:checked ~ .star-rating__label:hover ~ .star-rating__label,
+  .star-rating__label:hover ~ .star-rating__value:checked ~ .star-rating__label {
+    content: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="%23f0c808" class="size-6"><path fill-rule="evenodd" d="M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.006 5.404.434c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.434 2.082-5.005Z" clip-rule="evenodd" /></svg>');
```
Comment (fballiano) [inline public/skin/frontend/rwd/default/css/styles.css:1648]: question: why do we need all this svgs?

**Comment by empiricompany** [comment]
true, maybe it's better to pick the color orange used by Amazon?

File: public/skin/frontend/rwd/default/css/styles.css (hunk 1 above)
Comment (empiricompany) [reply public/skin/frontend/rwd/default/css/styles.css:1648]: Do you mean repetition? We can put the two SVGs in a CSS variable

Comment (fballiano) [reply public/skin/frontend/rwd/default/css/styles.css:1648]: I'd love that

**Comment by fballiano** [comment]
> true, maybe it's better to pick the color orange used by Amazon?

good idea, would it be possible to have the color in a css variable or something like that?

File: app/design/frontend/rwd/default/template/review/form.phtml [hunk 2]
```diff
@@ -31,7 +31,7 @@
                             <div class="star-rating">
                             <?php foreach (array_reverse($_rating->getOptions()) as $key=>$_option): ?>
                                 <input required class="star-rating__value" type="radio" id="<?php echo $this->escapeHtml($_rating->getRatingCode()) ?>_<?php echo $_option->getValue() ?>" value="<?php echo $_option->getId() ?>" name="ratings[<?php echo $_rating->getId() ?>]" value="<?php echo $_option->getId() ?>" />
-                                <label for="<?php echo $this->escapeHtml($_rating->getRatingCode()) ?>_<?php echo $_option->getValue() ?>" class="star-rating__label" title="<?php echo $_option->getValue(); ?>"><?php echo $_option->getValue(); ?></label>
+                                <label for="<?php echo $this->escapeHtml($_rating->getRatingCode()) ?>_<?php echo $_option->getValue() ?>" class="star-rating__label" title="<?php echo $this->__($_option->getValue() . ' star'); ?>"><?php echo $_option->getValue(); ?></label>
```
Comment (fballiano) [inline app/design/frontend/rwd/default/template/review/form.phtml:36]: genuine question: do you think this is worth adding 5 translations elements to the CSV?

File: public/skin/frontend/rwd/default/css/styles.css [hunk 3]
```diff
@@ -1574,8 +1574,10 @@ span.weee {
...
  * Ratings
  */
-.ratings {
+.ratings, .rating-box {
   margin: 7px 0;
+  --star-rating__icon-active: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="%23f0c808"><path fill-rule="evenodd" d="M10.788 3.21c.448-1.077 1.976-1.077 2.424 0l2.082 5.006 5.404.434c1.164.093 1.636 1.545.749 2.305l-4.117 3.527 1.257 5.273c.271 1.136-.964 2.033-1.96 1.425L12 18.354 7.373 21.18c-.996.608-2.231-.29-1.96-1.425l1.257-5.273-4.117-3.527c-.887-.76-.415-2.212.749-2.305l5.404-.434 2.082-5.005Z" clip-rule="evenodd" /></svg>');
+  --star-rating__icon-inactive: url('data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke-width="1.5" stroke="%23f0c808" class="size-6"><path stroke-linecap="round" stroke-linejoin="round" d="M11.48 3.499a.562.562 0 0 1 1.04 0l2.125 5.111a.563.563 0 0 0 .475.345l5.518.442c.499.04.701.663.321.988l-4.204 3.602a.563.563 0 0 0-.182.557l1.285 5.385a.562.562 0 0 1-.84.61l-4.725-2.885a.562.562 0 0 0-.586 0L6.982 20.54a.562.562 0 0 1-.84-.61l1.285-5.386a.562.562 0 0 0-.182-.557l-4.204-3.602a.562.562 0 0 1 .321-.988l5.518-.442a.563.563 0 0 0 .475-.345L11.48 3.5Z" /></svg>');
```
Comment (fballiano) [inline public/skin/frontend/rwd/default/css/styles.css:1580]: did you try the amazon orange color instead?

**Review by fballiano — APPROVED** [review]
Thanks @empiricompany, great changes :-)

File: app/design/frontend/rwd/default/template/review/form.phtml (hunk 2 above)
Comment (justinbeaty) [reply app/design/frontend/rwd/default/template/review/form.phtml:36]: It looks like those translations are already in both `Mage_Rating.csv` and `Mage_Review.csv`, but they should be "1 star", "2 stars", etc.

What about adding this method:

```php
class Mage_Rating_Model_Rating_Option extends Mage_Core_Model_Abstract
{
    // ... 

    public function getLabel(): string
    {
        if ($this->getValue() == 1) {
            return Mage::helper('rating')->__('%s star', $this->getValue());
        }
        return Mage::helper('rating')->__('%s stars', $this->getValue());
    }
}
```

And then:

```html
title="<?= $_option->getLabel(); ?>"
```

Comment (empiricompany) [reply app/design/frontend/rwd/default/template/review/form.phtml:36]: good idea! thanks

File: public/skin/frontend/rwd/default/css/styles.css (hunk 3 above)
Comment (empiricompany) [reply public/skin/frontend/rwd/default/css/styles.css:1580]: Yes, it's a matter of taste. It definitely has more contrast. It's not a problem to change it.
form
list

Comment (fballiano) [reply public/skin/frontend/rwd/default/css/styles.css]: loving this version!

**Comment by fballiano** [comment]
I think this is mergeable, @justinbeaty @empiricompany let's go?

**Comment by justinbeaty** [comment]
Yes I think it's a great change!

**Comment by fballiano** [comment]
done! thank you everybody!

**Pull request merged**